src/platform_impl/redox/mod.rs

```diff
@@ -0,0 +1,1336 @@
+    {
+        let exit_code = self.run_return(event_handler);
+        ::std::process::exit(exit_code);
+    }
+
+    pub fn run_return<F>(&mut self, mut event_handler_inner: F) -> i32
+    where
+        F: FnMut(event::Event<'_, T>, &event_loop::EventLoopWindowTarget<T>, &mut ControlFlow),
+    {
+        // Wrapper for event handler function that prevents ExitWithCode from being unset
+        let mut event_handler =
+            move |event: event::Event<'_, T>,
+                  window_target: &event_loop::EventLoopWindowTarget<T>,
+                  control_flow: &mut ControlFlow| {
+                if let ControlFlow::ExitWithCode(code) = control_flow {
+                    event_handler_inner(
+                        event,
+                        window_target,
+                        &mut ControlFlow::ExitWithCode(*code),
+                    );
+                } else {
+                    event_handler_inner(event, window_target, control_flow);
+                }
+            };
+
+        let mut control_flow = ControlFlow::default();
+        let mut start_cause = StartCause::Init;
+
+        let code = loop {
+            event_handler(
+                event::Event::NewEvents(start_cause),
+                &self.window_target,
+                &mut control_flow,
+            );
+
+            if start_cause == StartCause::Init {
+                event_handler(
+                    event::Event::Resumed,
+                    &self.window_target,
+                    &mut control_flow,
+                );
+            }
+
+            // Handle window creates
+            while let Some(window) = self.window_target.p.creates.lock().unwrap().pop_front() {
+                let window_id = WindowId {
+                    fd: window.fd as u64,
+                };
+
+                let mut buf: [u8; 4096] = [0; 4096];
+                let path = window.fpath(&mut buf).expect("failed to read properties");
+                let properties = WindowProperties::new(path);
+
+                self.windows.push(window);
+
+                // Send resize event on create to indicate first size
+                event_handler(
+                    event::Event::WindowEvent {
+                        window_id: window::WindowId(window_id),
+                        event: event::WindowEvent::Resized((properties.w, properties.h).into()),
+                    },
+                    &self.window_target,
+                    &mut control_flow,
+                );
+
+                // Send resize event on create to indicate first position
+                event_handler(
+                    event::Event::WindowEvent {
+                        window_id: window::WindowId(window_id),
+                        event: event::WindowEvent::Moved((properties.x, properties.y).into()),
+                    },
+                    &self.window_target,
+                    &mut control_flow,
+                );
```

Comment:
It could be done in `Window::new`, but it would involve additional allocations to put these events into the user events channel

Reply:
I'm going to say I vastly prefer the way it is right now compared to moving this to Window::new somehow. I could not simply use the user events channel since it only sends UserEvents right now. So I'd have to add more code in multiple places just to have these artificial events.